[addons][install] fix origin view direct after install of addon

AlwinEsch opened the pull request:
## Description

Before was if a new addon was installed and shown via "User installed add-ons" the origin (Repository) not known and shown as "Unknown", after a Kodi restart it was then available.

There was before after the install by CAddonInstallJob the manager called to add the new, but after the new was added there, the installer set the origin in database where was before not available in manager.
With this the database was correct and on next Kodi start used.

This change move the database update to the manager so that it also direct update the origin within his listed addons.

Ping @howie-f that's the fix about.

## User related

- Fix Repository show of new installed addon on his information dialog of "My add-ons", where was before shown as "Unknown" and after a Kodi restart available and correct.

> About Leia to Matrix no differences (as this fault was introduced in Matrix)

<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

**Text and images copied from https://github.com/xbmc/xbmc/pull/18181#issuecomment-671361033**

think i have come across another issue...

tested with a fresh install of kodi with https://github.com/xbmc/xbmc/pull/18181 applied:
1) start kodi and install the Southpark addon from our repo
2) next, go to `My Addons > Video addons > Southpark`
the origin is not shown correctly
![sp1](https://user-images.githubusercontent.com/687265/89786855-e2de2d00-db1c-11ea-83e1-f2d0660e3789.jpg)

3) restart Kodi and go to `My Addons > Video addons > Southpark` again
the origin is now shown correctly
![sp2](https://user-images.githubusercontent.com/687265/89786862-e5408700-db1c-11ea-98c7-1f108a07a54b.jpg)


## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

AlwinEsch commented on xbmc/addons/AddonManager.h:
```diff
@@ -331,6 +331,38 @@ namespace ADDON
+     * Currently listed call sources:
+     * - @ref CAddonInstallJob::DoWork
+     */
+    /*@{{{*/
```
About this think to look for cleanup of manager to have them better sorted and easier to see for where the are.

howie-f commented on xbmc/addons/AddonManager.h:
```diff
@@ -331,6 +331,38 @@ namespace ADDON
+     * @brief Update addon origin data.
+     *
+     * This becomes called from @ref CAddonInstallJob to set the source repo and
+     * if update, to set also the date about.
```
remove `about`

howie-f commented on xbmc/addons/AddonManager.h:
```diff
@@ -331,6 +331,38 @@ namespace ADDON
+     * This becomes called from @ref CAddonInstallJob to set the source repo and
+     * if update, to set also the date about.
+     *
+     * @note This must be called after the addon manager has with @ref FindAddons()
+     * inserted a new addon inside database.
```
`...after the addon manager has inserted a new addon with @ref FindAddons() into database.`

howie-f reviewed (approved):
just minor spelling (maybe our native english speakers will correct me though).

runtime test confirmed that this fixes the reported issue as well as in further places and improves the "available updates" - behavior (no restart needed anymore for available updates to show up)
well done!

ronie reviewed (approved):
i can also confirm this fixes the issue.

phunkyfish reviewed (approved):
LGTM, so just the comment updates from @howie-f that need updating.

AlwinEsch commented on xbmc/addons/AddonInstaller.cpp:
```diff
@@ -640,13 +640,10 @@ bool CAddonInstallJob::DoWork()
-  }
+  // Write now the origin data to database via addon manager, where also update
+  // present addon information, so that there new stored addons also becomes
+  // them in.
```
Thanks for the notes, yeah my english sometimes a bit stupid germlish :smile:

AlwinEsch committed:
[addons][install] fix origin view direct after install of addon

Before was if a new addon was installed and shown via "User installed add-ons"
the origin (Repository) not known and shown as "Unknown", after a Kodi restart
it was then available.

There was before after the install by CAddonInstallJob the manager called to add
the new, but after the new was added there, the installer set the origin
in database where was before not available in manager.
With this the database was correct and on next Kodi start used.

This change move the database update to the manager so that it also direct update
the origin within his listed addons.

AlwinEsch commented:
jenkins build this please
